src/components/Chart/MainChart.js

```diff
@@ -54,30 +54,32 @@ class MainChart extends Component {
     };
 
     initializeChart(dataset) {
-        // instantiate scenarios, initial default indicators
-        const SCENARIOS = buildScenarios(dataset);  
-        const scenarioList = SCENARIOS.map(s => s.name);
-        const scenarioMap = buildScenarioMap(dataset);
-        const statList = this.props.indicators.slice(0,2)
-
-        // instantiate start and end date (past 2 weeks) for summary indicators
-        const dates = dataset[SCENARIOS[0].key].dates.map( d => parseDate(d));
-        const start = new Date(); 
-        start.setDate(start.getDate() - 14); 
-
-        // dataset needs to be set to state at the same time as other props
-        // otherwise, children updates will occur at different times
-        this.setState({
-            datasetChart: dataset, 
-            dates,
-            SCENARIOS,
-            scenarioList,
-            scenarioMap,
-            statList,
-            start,
-        }, () => {
-            this.setState({dataLoaded: true});
-        })
+        if (Object.keys(dataset).length > 0) {
+            // instantiate scenarios, initial default indicators
+            const SCENARIOS = buildScenarios(dataset);  
+            const scenarioList = SCENARIOS.map(s => s.name);
+            const scenarioMap = buildScenarioMap(dataset);
+            const statList = this.props.indicators.slice(0,2)
+
+            // instantiate start and end date (past 2 weeks) for summary indicators
+            const dates = dataset[SCENARIOS[0].key].dates.map( d => parseDate(d));
+            const start = new Date(); 
+            start.setDate(start.getDate() - 14); 
+
+            // dataset needs to be set to state at the same time as other props
+            // otherwise, children updates will occur at different times
+            this.setState({
+                datasetChart: dataset, 
+                dates,
+                SCENARIOS,
+                scenarioList,
+                scenarioMap,
+                statList,
+                start,
+            }, () => {
+                this.setState({dataLoaded: true});
```

Comment:
note: This is not always going to be set after the previous setState call. It shouldn't matter for this case but it's good to know in case you try to do a setState callback that depends on state.

Reply:
I'm not sure if I've addressed this one? This is in Chart which only depends on `dataset`, so it should be fine?

Reply:
Would just put `dataLoaded` in the original setState and not a callback. Functionally it shouldn't matter but it makes it clearer.

Reply:
Oh, so like this instead... I use this callback after previous setState call in all of the containers (MainGraph, MainMap, MainChart, etc) - should they all be changed? You're saying that the callback doesn't even guarantee delayed set state so there's no point in even using one, correct?

``` this.setState({
                scenarios,
                indicators,
                dataLoaded: true
            });
```

Reply:
> You're saying that the callback doesn't even guarantee delayed set state so there's no point in even using one, correct?

Correct. The intention of the code and the result are not the same.